lib/facter/systemd.rb

```diff
@@ -0,0 +1,39 @@
+# Fact: systemd
+#
+# Purpose: 
+#   Determine whether SystemD is the init system on the node
+#
+# Resolution:
+#   Check the name of the process 1 (ps -p 1)
+#
+# Caveats:
+#
+
+# Fact: systemd-version
+#
+# Purpose: 
+#   Determine the version of systemd installed
+#
+# Resolution:
+#  Check the output of systemctl --version
+#
+# Caveats:
+#
+
+Facter.add(:systemd) do
+  confine :kernel => :linux
+  setcode do
+    result = false
+    init_process_name = Facter::Core::Execution.exec('ps -p 1 -o comm=')
+    if init_process_name.eql? 'systemd'
+      result = true
+    end
```

Comment:
I think you just want to do:

```
setcode do
   ...
   init_process_name == 'systemd'
end
```

In other words, the value of the fact should be the last value of the setcode block. If you're on a systemd system, then the last value is `true` due to evaluating `result = true`, e.g.

```
irb(main):001:0> result = true
=> true
```

If you're not on systemd, then the value is actually coming from the `if` check:

```
irb(main):002:0> if 1 == 2
irb(main):003:1> end
=> nil
```